site/src/pages/tutorials/tutorial-blog-rest-step3.mdx

```diff
@@ -0,0 +1,169 @@
+  ```java filename=BlogRequestConverter.java
+  final class BlogPostRequestConverter implements RequestConverterFunction {
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private final AtomicInteger idGenerator = new AtomicInteger(); // Blog post ID
+  }
+  ```
+3. Add a method retrieving a value of a given key in a JSON object:
+  ```java filename=BlogRequestConverter.java highlight="4"
+    static String stringValue(JsonNode jsonNode, String field) {
+      final JsonNode value = jsonNode.get(field);
+      if (value == null) {
+          throw new IllegalArgumentException(field + " is missing!");
+      }
+      return value.textValue();
+    }
+  ```
+4. Customize the default `convertRequest()` method as follows.
+    - Map the `title` and `content` properties of the body parameter to members in a `BlogPost` object, using the `stringValue()` method we wrote in the previous step.
+    - Throw the <type://FallthroughException> by calling <type://RequestConverterFunction#fallthrough()>, to pass on the request parameters that aren't converted by our converter to the default request converter.
+    - Feel free to add more mapping blocks (lines 9 to 11) in this request converter for other requests for the blog service.
+    ```java filename=BlogRequestConverter.java highlight=9,10,13
+    @Override
+    public Object convertRequest(ServiceRequestContext ctx,
+      AggregatedHttpRequest request, Class<?> expectedResultType,
+      @Nullable ParameterizedType expectedParameterizedResultType)
+        throws Exception {
+      if (expectedResultType == BlogPost.class) {
+        final JsonNode jsonNode = mapper.readTree(request.contentUtf8());
+        final int id = idGenerator.getAndIncrement();
+        final String title = stringValue(jsonNode, "title");
+        final String content = stringValue(jsonNode, "content");
+        return new BlogPost(id, title, content); // Create an instance of BlogPost object
+      }
+      return RequestConverterFunction.fallthrough();
+    }
+  ```
+
+### Assign a request converter
+
+In this step, assign the [request converter we customized](#write-a-request-converter) to our service method. Annotate the service method with <type://@RequestConverter> and specify the RequestConverterFunction class as `BlogPostRequestConverter.class`.
+
+```java filename=BlogService.java highlight=2
+@Post("/blogs")
+@RequestConverter(BlogPostRequestConverter.class)
+public HttpResponse createBlogPost(BlogPost blogPost) throws JsonProcessingException {
+  // Implement blog service
+}
+```
+
+## 3. Implement service code
+
+When the request for creation is received, our request converter creates an instance of a blog post object for us. We want to save the blog post object in the map (`blogPosts`) created in the `BlogService` class. Skip adding a database operation for this tutorial.
```

Comment:
I think we can remove the sentence `Skip adding ...`

Reply:
Done